File: 7KYU/Balanced Number (Special Numbers Series #1 )/solution.clj
(ns balanced-number)

(defn number->digits [num]
  (->> num str (map (comp read-string str))))

(defn balanced-num
  [num]
  (let [l (count (str num))
        digits (number->digits num)
        n (cond (even? l) (Math/ceil (/ l 2))
                :else (/ l 2))
        left (reduce + (take (dec n) digits))
        right (if (even? l)
                (reduce + (drop (inc n) digits))
                (reduce + (drop n digits)))]
    (if (= left right)
      "Balanced"
      "Not Balanced")))

File: 7KYU/Balanced Number (Special Numbers Series #1 )/solution.cs
using System.Linq;
using System;
class Kata
{
    public static string BalancedNumber(int number)
    {
      var l = number.ToString().Length;
      var digits = number.ToString().Select(d => (int) Char.GetNumericValue(d));
      int left;
      int right;

      if (l % 2 != 0)
      {
        var m = (int) Math.Ceiling(number.ToString().Length / 2.0);
        left = digits.Take(m - 1).Sum();
        right = digits.Skip(m).Sum();
      }
      else
      {
        var m = number.ToString().Length / 2;
        left = digits.Take(m - 1).Sum();
        right = digits.Skip(m + 1).Sum();
      }

      if (left == right)
        return "Balanced";
      return "Not Balanced";
    }
}